package/MDAnalysis/topology/ITPParser.py

```diff
@@ -215,16 +246,55 @@ def parse(self, **kwargs):
                        residue_segindex=segidx)
         
         # connectivity
-        for vals, Attr, attrname, atype in (
-            (bonds, Bonds, 'bonds', bondtypes),
-            (angles, Angles, 'angles', angletypes),
-            (dihedrals, Dihedrals, 'dihedrals', dihedraltypes),
-            (impropers, Impropers, 'impropers', impropertypes)
+        for dct, Attr, attrname in (
+            (bonds, Bonds, 'bonds'),
+            (angles, Angles, 'angles'),
+            (dihedrals, Dihedrals, 'dihedrals'),
+            (impropers, Impropers, 'impropers')
         ):
-            tattr = Attr(set(tuple(map(ids.index, x)) for x in vals),
-                         types=atype)
+            if dct:
+                indices, types = zip(*list(dct.items()))
+            else:
+                indices, types = [], []
+
+            types = [squash_identical(t) for t in types]
+            
+            tattr = Attr(indices, types=types)
             top.add_TopologyAttr(tattr)
 
         return top
+    
+    def skip_if(self):
+        """
+        Skip lines in if condition, *inclusive* of the #endif or #else.
+        """
+        line = next(self.itpfile)
```

Comment:
this is a fun example of where the walrus operator is actually useful (if we could jump straight to 3.8)